Sources/PackageModel/Manifest.swift

```diff
@@ -140,6 +151,37 @@ extension ToolsVersion {
 }
 
 extension Manifest {
+    /// Targets required for building all the products. If this manifest is a root manifest, it returns all targets.
+    public var allRequiredTargets: [TargetDescription] {
+        guard kind != .root else {
```

Comment:
please use `switch` so  we don't miss a case when a new one is added in the future